std/os.zig

```diff
@@ -100,67 +100,52 @@ pub const GetRandomError = OpenError;
 /// appropriate OS-specific library call. Otherwise it uses the zig standard
 /// library implementation.
 pub fn getrandom(buf: []u8) GetRandomError!void {
+    if (buf.len == 0) {
```

Comment:
I wouldn't bother with this check: what you often see is the check happening at every call-site which results in slow execution because of all the extra branches!
Plus, there's the rare situation that they called `getrandom` with an empty because they wanted to e.g. ensure that the system had entropy.